fastn-core/src/commands/serve.rs

```diff
@@ -349,6 +350,32 @@ pub async fn serve(
                 }
             }
 
+            if config.package.user_id {
+                let provider = "github";
+                if let Some(token) =
+                    fastn_core::auth::get_user_data_from_cookies(provider, "token", &cookies)
+                        .await
+                        .ok()
+                        .flatten()
+                {
+                    match fastn_core::auth::github::apis::user_details(token.as_str()).await {
```

Comment:
Making a http request for every request is expensive. Can you cache the user-id somewhere? A signed cookie is what we prefer.

Reply:
This, I think would be more secure than just passing `x-fastn-user-id` header to the proxy backend.

Reply:
I'd like to learn more about why you'd suggest using http header over a cookie. In my opinion, using a cookie would be much more convenient as we'd be able to remove all this code that creates custom header (`x-fastn-user-id`) if the cookie is present. And, caching wouldn't be needed if we store `user-id` as a signed cookie.

Reply:
My thinking it this: the header is better if the information is only passed from fastn to django, if it is a cookie then it should go till browser.